cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -103,15 +132,16 @@
   private final StreamCoordinatorClient streamCoordinatorClient;
   private final PrivilegesManager privilegesManager;
   private final AuthorizationEnforcer authorizationEnforcer;
+  private final Cache<ApplicationId, Injector> appInjectors;
 
   @Inject
-  PreviewHttpHandler(CConfiguration cConf, Configuration hConf, DiscoveryService discoveryService,
-                            DatasetFramework datasetFramework, PreferencesStore preferencesStore,
-                            SecureStore secureStore, TransactionManager transactionManager,
-                            ArtifactRepository artifactRepository, ArtifactStore artifactStore,
-                            AuthorizerInstantiator authorizerInstantiator, StreamAdmin streamAdmin,
-                            StreamCoordinatorClient streamCoordinatorClient, PrivilegesManager privilegesManager,
-                            AuthorizationEnforcer authorizationEnforcer) {
+  PreviewHttpHandler(final CConfiguration cConf, Configuration hConf, DiscoveryService discoveryService,
+                     @Named(DataSetsModules.BASE_DATASET_FRAMEWORK) DatasetFramework datasetFramework,
```

Comment:
Seems like we shouldn't have all these. The Http Handler should merely just a translation layer to translate REST call to internal Java method call (e.g. on PreviewService or PreviewManager). E.g. creation of new guice injection shouldn't be this class responsibility.